src/components/SearchBar.js

```diff
@@ -1,6 +1,12 @@
-import React from 'react';
+import React, { useRef, useState } from 'react';
 
 function SearchBar(props) {
+  const [searchValue, setSearchValue] = useState()
+  const searchInput = useRef()
+
+  function onChange(){
+    setSearchValue(searchInput.current.value)
```

Comment:
if you have `onChange` in the input then you don't need `ref` in the same input because it gives you absolutely the same reference to the element like `target` in the `event` which is provided by `onChange`.

And `useRef` is rarely used in React for inputs. Only when it's not possible to control an input with `onChange` and can't get `value` (in the input type=`file`) and when you need to `focus` an input when you open a popup or get to a page with a form.
To read more: https://reactjs.org/docs/uncontrolled-components.html#the-file-input-tag


```js

  function onChange(evt){
    setSearchValue(evt.target.value)

```